src/parsers/xyz.js

```diff
@@ -34,15 +34,52 @@ function validateLine(xyzLine, index) {
 }
 
 /**
- * Validates that passed string is well-formed XYZ file.
+ * Function checks the number of atoms listed in the BasisText editor.
+ * If the number of atoms > Basis.nonPeriodicMaxAtomsCount, then an error is thrown.
+ *
+ * @param {String} xyzTxt
+ */
+export function validateNumberOfAtoms(xyzTxt) {
+    const xyzArray = xyzTxt.split(/\r?\n/);
+    const nAtoms = xyzArray.length;
+    if (nAtoms > Basis.nonPeriodicMaxAtomsCount) {
+        throw new Error("Non-Periodic max atom limit exceeded");
+    }
+}
+
+/**
+ * Validates that passed string is well-formed XYZ file. Function returns an error code associated with a valid (0) or
+ * invalid (1001) format.
+ * @param xyzTxt {String}
+ * @return {Number}
+ */
+export function validateFormat(xyzTxt) {
+    try {
+        s(xyzTxt)
+            .trim()
+            .lines()
+            .filter((x) => x.trim() !== "")
+            .forEach(validateLine);
+    } catch (err) {
+        return 1001;
+    }
+    return 0;
```

Comment:
The logic for dealing with error codes should be in validateAll.

This function should just throw an Error like validateNumberOfAtoms above